webapp/channels/src/packages/mattermost-redux/src/actions/users.ts

```diff
@@ -278,14 +278,16 @@ export function getMissingProfilesByUsernames(usernames: string[]): ActionFunc {
     };
 }
 
-export function getProfilesByIds(userIds: string[], options?: any): ActionFunc {
+export function getProfilesByIds(userIds: string[], options?: any, includeCurrentUser = false): ActionFunc {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         const {currentUserId} = getState().entities.users;
         let profiles: UserProfile[];
 
         try {
             profiles = await Client4.getProfilesByIds(userIds, options);
-            removeUserFromList(currentUserId, profiles);
+            if (!includeCurrentUser) {
+                removeUserFromList(currentUserId, profiles);
```

Comment:
I also looked into the history of this (https://github.com/mattermost/mattermost-redux/pull/24), and it looks like the original reason for not doing this is because the user retrieved from this API doesn't include any of the private fields (like their email if that's hidden for other users). Since then, every one of our "fetch multiple users" endpoints has this same logic.

Coming back to this PR, if we include the current user then for this one case, it'll still overwrite those fields. Maybe instead of doing this, we should make it so that the reducer for `state.entities.users.profiles` merges the previously received user and the new one so that those extra fields are kept? Then the only change we'd need in this file is to remove `removeUserFromList` from this action entirely.

I'll also file a followup to remove the other uses of `removeUserFromList` as well since just merging the objects seems like a much better way to not lose the private fields while still letting us update the current user more easily. It'll also just make this code more straightforward since I would not have expected these actions would omit the current user

Reply:
Hi, thanks for the review ! I did some changes to the profiles reducer and removed the `removeUserFromList` from this action. Could you please tell me if this is what you thought about ? I can also submit a followup pr removing `removeUserFromList` in all profiles related actions if you want me to.  :)

Reply:
Yep, that's exactly what I was looking for. Would you mind also adding a unit test for the case where we have a user with an email and their notify_props set and then we receive a copy of that user without those fields just to make sure we don't accidentally break that in the future?

And that other PR would definitely be appreciated as well. I'll make a ticket on Jira so that we can fully track it on our end as well

Reply:
Sure, here is the unit test. I will try to submit the other pr as soon as I can.